src/js/menu.js

```diff
@@ -110,18 +164,39 @@ var showSensorsMenu = function(thermostat) {
       items: menuItems
     }]
   });
-  menu.show();
+  
+  hvacModeMenu.on('select', function(e) {
+    var postRequest = Utils.createChangeModeRequest(thermostat, e.item.value);
+    ecobeeApi.postThermostat(postRequest, 
+      function() {
+        hvacModeMenu.hide();
+        if (menu) menu.hide();
+      }, 
+      function(error) {
+        console.log('error setting hvac mode: '+error);
+    });
+  });
+  
+  hvacModeMenu.show();
 };
 
 this.exports = {
-  show: function(thermostat) {
-    if (menu) {
-      menu.hide();
-      menu = null;
+  show: function(thermostatList) {   
+    var thermostat;
+    var selectedThermostatId = Settings.data('selectedThermostatId');
+    if(selectedThermostatId){
+      thermostat = Utils.selectThermostat(selectedThermostatId,thermostatList);
+    }
+    else{
+      thermostat = thermostatList[0];
     }
+
     var menuItems = [];
     var hasHold = Utils.hasHold(thermostat);
     var hasSensors = Utils.hasSensors(thermostat);
+    if(thermostatList.length > 1){
+      menuItems.push({title: 'Thermostats'});
```

Comment:
I will change the order. I switch thermostats several times per day to gain access to the different sensors being in the sensor menu. I'm still trying to fine tune my comfort settings. But I imagine I'm not the typical user in this respect.